splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/PropertyConglomerate.java

```diff
@@ -126,7 +129,38 @@ public PropertyConglomerate(TransactionController tc, boolean create, Properties
 		PC_XenaVersion softwareVersion = new PC_XenaVersion();
 		if (create)
 			setProperty(tc,DataDictionary.PROPERTY_CONGLOMERATE_VERSION,softwareVersion, true);
-		getCachedProperties(tc);
+        boolean onHold = false;
+        Clock clock = SIDriver.driver().getClock();
+        do {
+            try {
+                onHold = false;
+		        getCachedProperties(tc);
+            }  catch (Exception e1) {
+                // The exception has so many levels, so use a loop to get the cause
+                Throwable cause = e1;
+                while (cause.getCause() != null) {
+                    String className = cause.getCause().getClass().getName();
+                    // Cannot get the origin class because of the dependency, so compare them by
+                    // class name
+                    if (className.equals("org.apache.hadoop.hbase.NotServingRegionException") ||
+                            className.equals("org.apache.hadoop.hbase.exceptions.RegionOpeningException")) {
+                        onHold = true;
+                        break;
+                    }
+                    cause = cause.getCause();
+                }
+                if (onHold) {
+                    try {
+                        clock.sleep(1, TimeUnit.SECONDS);
```

Comment:
Since this is an infinite wait, can you log a message to let the user know what's going on if the system is not coming up? Otherwise it looks good to me!

Reply:
@dgomezferro Added the log